src/cpp/rtps/security/SecurityManager.cpp

```diff
@@ -123,12 +124,95 @@ bool SecurityManager::init(
     security_activated = false;
 
     SecurityException exception;
+
     domain_id_ = participant_->getRTPSParticipantAttributes().builtin.domainId;
 
+    const PropertyPolicy log_properties = PropertyPolicyHelper::get_properties_with_prefix(
+          participant_->getRTPSParticipantAttributes().properties,
+          "dds.sec.log.builtin.DDS_LogTopic.");
+
+    // length(log_properties) == 0 considered as logging disable.
+    if (PropertyPolicyHelper::length(log_properties) > 0)
+    {
+      logging_plugin_ = factory_.create_logging_plugin(participant_properties);
+
+      if (logging_plugin_ != nullptr)
+      {
+        LogOptions log_options;
+        log_options.distribute = true; //TODO(artivis): defaults to true until log to file is implemented.
+        log_options.log_level = LoggingLevel::ERROR_LEVEL;
+        log_options.log_file = "";
+
+        const auto init_logging_fail = [&](){
```

Comment:
I think it would be more clear to accept the exception as an arg here instead of capturing it.

Reply:
Capture the `logging_plugin_` pointer but pass the exception? Not sure that's really better.

Reply:
Well the logging plugin was defined above and doesn't change, whereas in order for this function to do anything meaningful the caller must first define the exception and THEN call this function, which seems like a mistake waiting to happen.